src/lib/keccak.ts

```diff
@@ -64,14 +102,156 @@ const ROUND_CONSTANTS = [
   0x8000000000008080n,
   0x0000000080000001n,
   0x8000000080008008n,
-].map(Field.from);
+];
+
+// AUXILARY FUNCTIONS
+
+// Auxiliary function to check composition of 8 bytes into a 64-bit word
+function checkBytesToWord(word: Field, wordBytes: Field[]): void {
+  let composition = wordBytes.reduce((acc, x, i) => {
+    const shift = Field.from(2n ** BigInt(8 * i));
+    return acc.add(x.mul(shift));
+  }, Field.from(0));
+
+  word.assertEquals(composition);
+}
+
+// KECCAK STATE FUNCTIONS
 
 // Return a keccak state where all lanes are equal to 0
 const getKeccakStateZeros = (): Field[][] =>
   Array.from(Array(KECCAK_DIM), (_) => Array(KECCAK_DIM).fill(Field.from(0)));
 
+// Converts a list of bytes to a matrix of Field elements
```

Comment:
This one does not create constraints for the byte length? Asking because `keccakStateToBytes` does. I guess this is because `getKeccakStateOfBytes` is only used at the beginning of Keccak where inputs are already constraints, but `keccakStateToBytes` is called at the end where you might want to check fresh bytes of the hash output?